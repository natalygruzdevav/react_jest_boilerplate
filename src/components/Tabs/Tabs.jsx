import React, { Component } from 'react';
import {
  Tab, Tabs, TabList, TabPanel,
} from 'react-tabs';

import 'react-tabs/style/react-tabs.css';

class TabsCоmponent extends Component {
  render() {
    return (
      <Tabs>
        <TabList>
          <Tab>Title 1</Tab>
          <Tab disabled>Title 2</Tab>
          <Tab>Title 3</Tab>
          <Tab>Title 4</Tab>
          <Tab>Title 5</Tab>
          <Tab>Title 6</Tab>
        </TabList>

        <TabPanel>
          <h2>Any content 1</h2>
        </TabPanel>
        <TabPanel>
          <h2>Any content 2</h2>
        </TabPanel>
        <TabPanel>
          <h2>Any content 3</h2>
        </TabPanel>
        <TabPanel>
          <h2>Any content 4</h2>
        </TabPanel>
        <TabPanel>
          <h2>Any content 5</h2>
        </TabPanel>
        <TabPanel>
          <h2>Any content 6</h2>
        </TabPanel>
      </Tabs>
    );
  }
}

export default TabsCоmponent;
