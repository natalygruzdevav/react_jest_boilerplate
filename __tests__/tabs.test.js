import React from 'react';
import { mount, shallow } from 'enzyme';

import Tabs from '../src/components/Tabs';

describe('Tabs', () => {
  it('renders', () => {
    const wrapper = shallow(<Tabs />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render selected tab', () => {
    const wrapper = mount(<Tabs default={4} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('select tab', () => {
    const wrapper = mount(<Tabs />);
    wrapper
      .find('Tab')
      .at(5)
      .simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('not react on disabled tab', () => {
    const wrapper = mount(<Tabs />);

    wrapper.find({ disabled: true }).simulate('click');
    expect(wrapper).toMatchSnapshot();

    wrapper
      .find('Tab')
      .at(0)
      .simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
