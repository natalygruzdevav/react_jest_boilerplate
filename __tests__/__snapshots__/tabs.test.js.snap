// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs not react on disabled tab 1`] = `
<TabsCоmponent>
  <Tabs
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    selectedIndex={null}
  >
    <UncontrolledTabs
      className="react-tabs"
      focus={false}
      forceRenderTabPanel={false}
      onSelect={[Function]}
      selectedIndex={0}
    >
      <div
        className="react-tabs"
        data-tabs={true}
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <TabList
          className="react-tabs__tab-list"
          key=".0"
        >
          <ul
            className="react-tabs__tab-list"
            role="tablist"
          >
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-24"
              key=".0"
              panelId="react-tabs-25"
              selected={true}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-25"
                aria-disabled="false"
                aria-selected="true"
                className="react-tabs__tab react-tabs__tab--selected"
                id="react-tabs-24"
                role="tab"
                tabIndex="0"
              >
                Title 1
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabled={true}
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-26"
              key=".1"
              panelId="react-tabs-27"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-27"
                aria-disabled="true"
                aria-selected="false"
                className="react-tabs__tab react-tabs__tab--disabled"
                id="react-tabs-26"
                role="tab"
                tabIndex={null}
              >
                Title 2
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-28"
              key=".2"
              panelId="react-tabs-29"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-29"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-28"
                role="tab"
                tabIndex={null}
              >
                Title 3
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-30"
              key=".3"
              panelId="react-tabs-31"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-31"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-30"
                role="tab"
                tabIndex={null}
              >
                Title 4
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-32"
              key=".4"
              panelId="react-tabs-33"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-33"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-32"
                role="tab"
                tabIndex={null}
              >
                Title 5
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-34"
              key=".5"
              panelId="react-tabs-35"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-35"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-34"
                role="tab"
                tabIndex={null}
              >
                Title 6
              </li>
            </Tab>
          </ul>
        </TabList>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-25"
          key=".1"
          selected={true}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-24"
        >
          <div
            aria-labelledby="react-tabs-24"
            className="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="react-tabs-25"
            role="tabpanel"
          >
            <h2>
              Any content 1
            </h2>
          </div>
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-27"
          key=".2"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-26"
        >
          <div
            aria-labelledby="react-tabs-26"
            className="react-tabs__tab-panel"
            id="react-tabs-27"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-29"
          key=".3"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-28"
        >
          <div
            aria-labelledby="react-tabs-28"
            className="react-tabs__tab-panel"
            id="react-tabs-29"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-31"
          key=".4"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-30"
        >
          <div
            aria-labelledby="react-tabs-30"
            className="react-tabs__tab-panel"
            id="react-tabs-31"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-33"
          key=".5"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-32"
        >
          <div
            aria-labelledby="react-tabs-32"
            className="react-tabs__tab-panel"
            id="react-tabs-33"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-35"
          key=".6"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-34"
        >
          <div
            aria-labelledby="react-tabs-34"
            className="react-tabs__tab-panel"
            id="react-tabs-35"
            role="tabpanel"
          />
        </TabPanel>
      </div>
    </UncontrolledTabs>
  </Tabs>
</TabsCоmponent>
`;

exports[`Tabs not react on disabled tab 2`] = `
<TabsCоmponent>
  <Tabs
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    selectedIndex={null}
  >
    <UncontrolledTabs
      className="react-tabs"
      focus={false}
      forceRenderTabPanel={false}
      onSelect={[Function]}
      selectedIndex={0}
    >
      <div
        className="react-tabs"
        data-tabs={true}
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <TabList
          className="react-tabs__tab-list"
          key=".0"
        >
          <ul
            className="react-tabs__tab-list"
            role="tablist"
          >
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-24"
              key=".0"
              panelId="react-tabs-25"
              selected={true}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-25"
                aria-disabled="false"
                aria-selected="true"
                className="react-tabs__tab react-tabs__tab--selected"
                id="react-tabs-24"
                role="tab"
                tabIndex="0"
              >
                Title 1
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabled={true}
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-26"
              key=".1"
              panelId="react-tabs-27"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-27"
                aria-disabled="true"
                aria-selected="false"
                className="react-tabs__tab react-tabs__tab--disabled"
                id="react-tabs-26"
                role="tab"
                tabIndex={null}
              >
                Title 2
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-28"
              key=".2"
              panelId="react-tabs-29"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-29"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-28"
                role="tab"
                tabIndex={null}
              >
                Title 3
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-30"
              key=".3"
              panelId="react-tabs-31"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-31"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-30"
                role="tab"
                tabIndex={null}
              >
                Title 4
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-32"
              key=".4"
              panelId="react-tabs-33"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-33"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-32"
                role="tab"
                tabIndex={null}
              >
                Title 5
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-34"
              key=".5"
              panelId="react-tabs-35"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-35"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-34"
                role="tab"
                tabIndex={null}
              >
                Title 6
              </li>
            </Tab>
          </ul>
        </TabList>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-25"
          key=".1"
          selected={true}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-24"
        >
          <div
            aria-labelledby="react-tabs-24"
            className="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="react-tabs-25"
            role="tabpanel"
          >
            <h2>
              Any content 1
            </h2>
          </div>
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-27"
          key=".2"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-26"
        >
          <div
            aria-labelledby="react-tabs-26"
            className="react-tabs__tab-panel"
            id="react-tabs-27"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-29"
          key=".3"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-28"
        >
          <div
            aria-labelledby="react-tabs-28"
            className="react-tabs__tab-panel"
            id="react-tabs-29"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-31"
          key=".4"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-30"
        >
          <div
            aria-labelledby="react-tabs-30"
            className="react-tabs__tab-panel"
            id="react-tabs-31"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-33"
          key=".5"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-32"
        >
          <div
            aria-labelledby="react-tabs-32"
            className="react-tabs__tab-panel"
            id="react-tabs-33"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-35"
          key=".6"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-34"
        >
          <div
            aria-labelledby="react-tabs-34"
            className="react-tabs__tab-panel"
            id="react-tabs-35"
            role="tabpanel"
          />
        </TabPanel>
      </div>
    </UncontrolledTabs>
  </Tabs>
</TabsCоmponent>
`;

exports[`Tabs render selected tab 1`] = `
<TabsCоmponent
  default={4}
>
  <Tabs
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    selectedIndex={null}
  >
    <UncontrolledTabs
      className="react-tabs"
      focus={false}
      forceRenderTabPanel={false}
      onSelect={[Function]}
      selectedIndex={0}
    >
      <div
        className="react-tabs"
        data-tabs={true}
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <TabList
          className="react-tabs__tab-list"
          key=".0"
        >
          <ul
            className="react-tabs__tab-list"
            role="tablist"
          >
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-0"
              key=".0"
              panelId="react-tabs-1"
              selected={true}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-1"
                aria-disabled="false"
                aria-selected="true"
                className="react-tabs__tab react-tabs__tab--selected"
                id="react-tabs-0"
                role="tab"
                tabIndex="0"
              >
                Title 1
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabled={true}
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-2"
              key=".1"
              panelId="react-tabs-3"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-3"
                aria-disabled="true"
                aria-selected="false"
                className="react-tabs__tab react-tabs__tab--disabled"
                id="react-tabs-2"
                role="tab"
                tabIndex={null}
              >
                Title 2
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-4"
              key=".2"
              panelId="react-tabs-5"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-5"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-4"
                role="tab"
                tabIndex={null}
              >
                Title 3
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-6"
              key=".3"
              panelId="react-tabs-7"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-7"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-6"
                role="tab"
                tabIndex={null}
              >
                Title 4
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-8"
              key=".4"
              panelId="react-tabs-9"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-9"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-8"
                role="tab"
                tabIndex={null}
              >
                Title 5
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-10"
              key=".5"
              panelId="react-tabs-11"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-11"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-10"
                role="tab"
                tabIndex={null}
              >
                Title 6
              </li>
            </Tab>
          </ul>
        </TabList>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-1"
          key=".1"
          selected={true}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-0"
        >
          <div
            aria-labelledby="react-tabs-0"
            className="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="react-tabs-1"
            role="tabpanel"
          >
            <h2>
              Any content 1
            </h2>
          </div>
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-3"
          key=".2"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-2"
        >
          <div
            aria-labelledby="react-tabs-2"
            className="react-tabs__tab-panel"
            id="react-tabs-3"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-5"
          key=".3"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-4"
        >
          <div
            aria-labelledby="react-tabs-4"
            className="react-tabs__tab-panel"
            id="react-tabs-5"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-7"
          key=".4"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-6"
        >
          <div
            aria-labelledby="react-tabs-6"
            className="react-tabs__tab-panel"
            id="react-tabs-7"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-9"
          key=".5"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-8"
        >
          <div
            aria-labelledby="react-tabs-8"
            className="react-tabs__tab-panel"
            id="react-tabs-9"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-11"
          key=".6"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-10"
        >
          <div
            aria-labelledby="react-tabs-10"
            className="react-tabs__tab-panel"
            id="react-tabs-11"
            role="tabpanel"
          />
        </TabPanel>
      </div>
    </UncontrolledTabs>
  </Tabs>
</TabsCоmponent>
`;

exports[`Tabs renders 1`] = `
<Tabs
  defaultFocus={false}
  defaultIndex={null}
  forceRenderTabPanel={false}
  selectedIndex={null}
>
  <TabList
    className="react-tabs__tab-list"
  >
    <Tab
      className="react-tabs__tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Title 1
    </Tab>
    <Tab
      className="react-tabs__tab"
      disabled={true}
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Title 2
    </Tab>
    <Tab
      className="react-tabs__tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Title 3
    </Tab>
    <Tab
      className="react-tabs__tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Title 4
    </Tab>
    <Tab
      className="react-tabs__tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Title 5
    </Tab>
    <Tab
      className="react-tabs__tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Title 6
    </Tab>
  </TabList>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <h2>
      Any content 1
    </h2>
  </TabPanel>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <h2>
      Any content 2
    </h2>
  </TabPanel>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <h2>
      Any content 3
    </h2>
  </TabPanel>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <h2>
      Any content 4
    </h2>
  </TabPanel>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <h2>
      Any content 5
    </h2>
  </TabPanel>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <h2>
      Any content 6
    </h2>
  </TabPanel>
</Tabs>
`;

exports[`Tabs select tab 1`] = `
<TabsCоmponent>
  <Tabs
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    selectedIndex={null}
  >
    <UncontrolledTabs
      className="react-tabs"
      focus={false}
      forceRenderTabPanel={false}
      onSelect={[Function]}
      selectedIndex={5}
    >
      <div
        className="react-tabs"
        data-tabs={true}
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <TabList
          className="react-tabs__tab-list"
          key=".0"
        >
          <ul
            className="react-tabs__tab-list"
            role="tablist"
          >
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-12"
              key=".0"
              panelId="react-tabs-13"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-13"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-12"
                role="tab"
                tabIndex={null}
              >
                Title 1
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabled={true}
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-14"
              key=".1"
              panelId="react-tabs-15"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-15"
                aria-disabled="true"
                aria-selected="false"
                className="react-tabs__tab react-tabs__tab--disabled"
                id="react-tabs-14"
                role="tab"
                tabIndex={null}
              >
                Title 2
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-16"
              key=".2"
              panelId="react-tabs-17"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-17"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-16"
                role="tab"
                tabIndex={null}
              >
                Title 3
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-18"
              key=".3"
              panelId="react-tabs-19"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-19"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-18"
                role="tab"
                tabIndex={null}
              >
                Title 4
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-20"
              key=".4"
              panelId="react-tabs-21"
              selected={false}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-21"
                aria-disabled="false"
                aria-selected="false"
                className="react-tabs__tab"
                id="react-tabs-20"
                role="tab"
                tabIndex={null}
              >
                Title 5
              </li>
            </Tab>
            <Tab
              className="react-tabs__tab"
              disabledClassName="react-tabs__tab--disabled"
              focus={false}
              id="react-tabs-22"
              key=".5"
              panelId="react-tabs-23"
              selected={true}
              selectedClassName="react-tabs__tab--selected"
              tabRef={[Function]}
            >
              <li
                aria-controls="react-tabs-23"
                aria-disabled="false"
                aria-selected="true"
                className="react-tabs__tab react-tabs__tab--selected"
                id="react-tabs-22"
                role="tab"
                tabIndex="0"
              >
                Title 6
              </li>
            </Tab>
          </ul>
        </TabList>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-13"
          key=".1"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-12"
        >
          <div
            aria-labelledby="react-tabs-12"
            className="react-tabs__tab-panel"
            id="react-tabs-13"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-15"
          key=".2"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-14"
        >
          <div
            aria-labelledby="react-tabs-14"
            className="react-tabs__tab-panel"
            id="react-tabs-15"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-17"
          key=".3"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-16"
        >
          <div
            aria-labelledby="react-tabs-16"
            className="react-tabs__tab-panel"
            id="react-tabs-17"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-19"
          key=".4"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-18"
        >
          <div
            aria-labelledby="react-tabs-18"
            className="react-tabs__tab-panel"
            id="react-tabs-19"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-21"
          key=".5"
          selected={false}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-20"
        >
          <div
            aria-labelledby="react-tabs-20"
            className="react-tabs__tab-panel"
            id="react-tabs-21"
            role="tabpanel"
          />
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          id="react-tabs-23"
          key=".6"
          selected={true}
          selectedClassName="react-tabs__tab-panel--selected"
          tabId="react-tabs-22"
        >
          <div
            aria-labelledby="react-tabs-22"
            className="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="react-tabs-23"
            role="tabpanel"
          >
            <h2>
              Any content 6
            </h2>
          </div>
        </TabPanel>
      </div>
    </UncontrolledTabs>
  </Tabs>
</TabsCоmponent>
`;
